### DISCUSSION TITLE
How do you arrange the ikvmstub step in <IkvmReference>?

### DISCUSSION THREAD

**Discussion by Kasper-Sumatra** [ikvmnet/ikvm#414] [Q&A]
I'm trying to upgrade an old IKVM project (a class library) to output an assembly targeting .NET 4.8 (and in the future maybe .NET 6). The project was built using IKVM 0.34.0.3, which does not seem to support that. So I'm trying to use the 'new way', where you just change your project reference to SDK style, and import the NuGet package, and add the \<IkvmReference\>. However, my project doesn't just have a .jar file. The java code references .NET code because half of the project source code was actually ported to c# at some point. In the old .csproj file I have 3 steps: 
 1- run ikvmstub on the c# library (and mscorlib) to generate .jar files for the java code to reference
 2- javac the original .java files and the just generated .jar files
 3- turn the resulting .jar file into a dll

A toy example I started with first works great; I can turn a .jar into a dll. So it seems step 3 works fine using the new \<IkvmReference\>, but I've no clue how to go about doing step 1 and 2. Can anyone point me in the right direction? Or should I stick to the old .csproj structure with the 3 steps, and just update the IKVM assemblies / executables?

**Comment by wasabii** [comment]
https://www.nuget.org/packages/IKVM.NET.Sdk/8.6.2

Try that.
